src/azure-cli/azure/cli/command_modules/managedservices/custom.py

```diff
@@ -0,0 +1,219 @@
+        Authorization
+
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+
+    if not definition_id:
+        definition_id = str(uuid.uuid4())
+
+    sub_id = _get_subscription_id(cmd, subscription)
+    scope = _get_scope(sub_id)
+
+    if plan_name and plan_product and plan_publisher and plan_version:
+        plan = Plan(plan_name, plan_publisher, plan_product, plan_version)
+    else:
+        plan = None
+
+    authorization = Authorization(
+        principal_id=principal_id,
+        role_definition_id=role_definition_id)
+
+    authorizations = [authorization]
+
+    properties = RegistrationDefinitionProperties(
+        description=description,
+        authorizations=authorizations,
+        registration_definition_name=name,
+        managed_by_tenant_id=tenant_id,
+        Plan=plan,
+    )
+
+    return client.create_or_update(
+        registration_definition_id=definition_id,
+        api_version=_get_api_version(api_version),
+        scope=scope,
+        properties=properties)
+
+
+# pylint: disable=unused-argument
+def cli_definition_get(cmd, client,
+                       definition,
+                       **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+    definition_id, sub_id, rg_name = _get_resource_id_parts(cmd, definition, subscription)
+    scope = _get_scope(sub_id, rg_name)
+    return client.get(
+        scope=scope,
+        registration_definition_id=definition_id,
+        api_version=_get_api_version(api_version))
+
+
+# pylint: disable=unused-argument
+def cli_definition_list(cmd, client, **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+    sub_id = _get_subscription_id(cmd, subscription)
+    scope = _get_scope(sub_id)
+    return client.list(
+        scope=scope,
+        api_version=_get_api_version(api_version))
+
+
+def cli_definition_delete(cmd, client,
+                          definition, **kwargs):
+    subscription = kwargs.pop('subscription', None)
+    api_version = kwargs.pop('api_version', None)
+    definition_id, sub_id, rg_name = _get_resource_id_parts(cmd, definition, subscription)
+    scope = _get_scope(sub_id, rg_name)
+    return client.delete(
+        scope=scope,
+        registration_definition_id=definition_id,
+        api_version=_get_api_version(api_version))
+
+
+# endregion
+
+# region Assignments Custom Commands
+
+# pylint: disable=unused-argument
+def cli_assignment_create(cmd, client,
+                          definition_id,
+                          assignment_id=None,
+                          **kwargs):
+    from azure.mgmt.managedservices.models import RegistrationAssignmentProperties
+    if not is_valid_resource_id(definition_id):
+        raise ValueError(
+            "registration_definition_resource_id should be a valid resource id. For example, "
```

Comment:
"registration_definition_resource_id" should correspond to the actual option, i.e. `--definition-id` or whatever it is.

Reply:
changed to definition